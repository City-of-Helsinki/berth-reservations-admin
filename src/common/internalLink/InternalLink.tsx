import React from 'react';
import { Link, LinkProps } from 'react-router-dom';
import classNames from 'classnames';

import styles from './internalLink.module.scss';

type Props = {
  color?: 'standard' | 'critical' | 'brand';
  underlined?: boolean;
  icon?: JSX.Element;
} & LinkProps;

const InternalLink: React.SFC<Props> = ({
  color = 'brand',
  icon,
  children,
  underlined,
  ...props
}) => {
  return (
    <Link
      {...props}
      className={classNames(styles.internalLink, styles[color], {
        [styles.underlined]: underlined,
      })}
    >
      {children}
    </Link>
  );
};

export default InternalLink;
