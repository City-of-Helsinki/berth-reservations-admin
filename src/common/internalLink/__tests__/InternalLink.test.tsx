import React from 'react';
import { shallow, mount } from 'enzyme';
import { HashRouter, Link } from 'react-router-dom';

import InternalLink, { InternalLinkProps } from '../InternalLink';

describe('InternalLink', () => {
  const getWrapper = (props?: Omit<InternalLinkProps, 'to'>) =>
    shallow(
      <HashRouter>
        <InternalLink to="/" children="Home Page" {...props} />
      </HashRouter>
    );

  it('renders normally', () => {
    const wrapper = getWrapper();

    expect(wrapper.render()).toMatchSnapshot();
  });

  it('renders normally with element children', () => {
    const wrapper = getWrapper({ children: <p>Test!</p> });

    expect(wrapper.render()).toMatchSnapshot();
  });

  it('renders Link component', () => {
    const wrapper = mount(
      <HashRouter>
        <InternalLink to="/">foo</InternalLink>
      </HashRouter>
    );

    expect(wrapper.find(Link)).toHaveLength(1);
  });
});
