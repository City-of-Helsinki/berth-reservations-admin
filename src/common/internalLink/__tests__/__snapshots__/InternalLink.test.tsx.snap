// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InternalLink renders normally 1`] = `
<a
  class="internalLink brand"
  href="#/"
  title="Home Page"
>
  Home Page
</a>
`;

exports[`InternalLink renders normally with element children 1`] = `
<a
  class="internalLink brand"
  href="#/"
>
  <p>
    Test!
  </p>
</a>
`;
